### ISSUE TITLE
Gridstack stopped working in version 2.0.0 and above if the `class="grid-stack"` element is a web-component with shadow-dom!

### ISSUE THREAD

**Issue by bhutiyakishan1** [gridstack/gridstack.js#1438]
## Gridstack stopped working in version 2.0.0 and above if the `class="grid-stack"` element is a web-component with shadow-dom!
 
I was on version grid-stack 1.2.0 before, and i am using it inside a web-component where the top most element(class="grid-stack) is a web-component and has a shadow-dom, it was working perfectly until i moved to version 2.0.0! Also tried checking the latest, 2.1.0! But with that version it doesn't work either! 

## Your environment
* 2.0.0 and above
* Chrome latest

## Steps to reproduce

```html
<vi-dashboard id="dashboardEl" class="grid-stack">
  <vi-panel loc-x="0" loc-y="0" width="3" height="3" class="grid-stack-item">
       <div class="grid-stack-item-content></div>
  </vi-panel>
  <vi-panel loc-x="3" loc-y="0" width="3" height="3" class="grid-stack-item>
       <div class="grid-stack-item-content></div>
  </vi-panel>
</vi-dashboard>
```
If you see above markup, vi-panel and vi-dashboard are both web-components! vi-panel being a webcomponent works fine! But if vi-dashboard is a web-component, it doesnt render properly! With no errors in console!

## Expected behaviour
It should render properly as normal grid-stack demo! 
![Screen Shot 2020-11-03 at 3 06 16 PM](https://user-images.githubusercontent.com/21228925/98035306-4c9d6880-1de6-11eb-844d-f5d45d56135e.png)

## Actual behaviour
Everything is stacked at one place ! Drag-drop and resizing doesn't work either!

![Screen Shot 2020-11-03 at 3 05 55 PM](https://user-images.githubusercontent.com/21228925/98035292-47401e00-1de6-11eb-8a3a-ecbc82769633.png)

**Comment by adumesny** [comment]
looks like a CSS issue to me and rows issues which id dynamically created... did you try `GridStack.init({styleInHead: true},...)` and see if that changes things ? (by default it puts the css right before the grid - in same parent.
if Not then I would step into `_updateStyles()` and see if it's able to create the CSS file and where it goes...

Also you didn't post a reproduceable case - would be nice to have a small zip project - so I can debug...

**Comment by bhutiyakishan1** [comment]
Reproduced the issue: https://codepen.io/bhutiyakishan1/pen/bGeKvgM

**Comment by altonfroikin** [comment]
I am seeing this issue as well in my polymer project as @bhutiyakishan1 showed in his demo.

**Comment by adumesny** [comment]
v3 example showing issue:
```js
<html>
<head>
  <!-- Polyfills only needed for Firefox and Edge. -->
  <script src="https://unpkg.com/@webcomponents/webcomponentsjs@latest/webcomponents-loader.js"></script>
  <link rel="stylesheet" href="demo.css"/>
  <script src="../dist/gridstack-h5.js"></script>
</head>
<body>
  <script type="module">
    import {LitElement, html, css} from 'https://unpkg.com/lit-element/lit-element.js?module';
    
    class MyElement extends LitElement {
      static get properties() { return {} }
      render() {
        return html`<style>:host {display: block;} </style><slot></slot>`;
      }
    }
    customElements.define('my-element', MyElement);
</script>

<my-element class="grid-stack"></my-element>

<script type="text/javascript">
  let items = [
    {x:0, y:0, w:2, content: 0},
    {x:0, y:1, content: 1}
  ];

  let grid = GridStack.init();
  grid.load(items);
</script>
</body>
</html>
```

**Comment by adumesny** [comment]
@bhutiyakishan1  found the issue... apparently we get size 0 for the grid and fail to load/create the needed CSS. Doing this for now should fix your issues:
```js
  let grid = GridStack.init({disableOneColumnMode: true, cellHeight: 70});
```

Note: I don't understand what this does in your code and how it's supposed to be used (changing to inline-block didn't fix the issue of grid not having a size)
```js
render() { return html`<style>:host {display: block;} </style><slot></slot>`; }
```

**Comment by adumesny** [comment]
fixed in 3.1.1

**Comment by bhutiyakishan1** [comment]
Thank you ! Appreciate the help !
